Update Player Constructor

stephensmeal opened the pull request:
player.py - Added logic to constructor ensuring a Suspect Card is
   passed as the parameter, if not a ValueError is raised.
test_player.py - Edited constructor test to check that new Card
   validation logic is correct.

stephensmeal committed:
Update Player Constructor

player.py - Added logic to constructor ensuring a Suspect Card is
   passed as the parameter, if not a ValueError is raised.
test_player.py - Edited constructor test to check that new Card
   validation logic is correct.

yo09975 commented on src/player.py:
```diff
@@ -28,14 +29,17 @@ class Player(object):
-        self._current_location = None
-        self._previous_location = None
-        self._was_transferred = False
+        if card.get_type() == CardType.SUSPECT:
```
Can you just make this:

if not card.get_type() == CardType.SUSPECT:
raise ValueError...

It's much cleaner that way. Use git commit --amend vs adding a new commit for this update.

yo09975 reviewed (changes requested):
After amending the first commit, add a new commit to fix the playerlist test that fails due to attempting to create a player with a ROOM card.

stephensmeal committed:
Fixed PlayerList Unit Tests

test_playerlist.py - Updated test Cards with Suspect CardType to work
   with new Player constructor data validation.
